holidays/countries/switzerland.py

```diff
@@ -9,227 +9,267 @@
-
-        if self.subdiv in {
-            "AI",
-            "JU",
-            "LU",
-            "NW",
-            "OW",
-            "SZ",
-            "TI",
-            "UR",
-            "VS",
-            "ZG",
-        }:
-            self[date(year, AUG, 15)] = "Mariä Himmelfahrt"
-
-        if self.subdiv == "VD":
-            # Monday after the third Sunday of September
-            dt = _get_nth_weekday_of_month(3, SUN, SEP, year) + td(days=+1)
-            self[dt] = "Lundi du Jeûne"
-
-        if self.subdiv == "GE":
-            # Thursday after the first Sunday of September
-            dt = _get_nth_weekday_of_month(1, SUN, SEP, year) + td(days=+4)
-            self[dt] = "Jeûne genevois"
-
-        if self.subdiv == "OW":
-            self[date(year, SEP, 25)] = "Bruder Klaus"
-
-        if self.subdiv in {
-            "AI",
-            "GL",
-            "JU",
-            "LU",
-            "NW",
-            "OW",
-            "SG",
-            "SZ",
-            "TI",
-            "UR",
-            "VS",
-            "ZG",
-        }:
-            self[date(year, NOV, 1)] = "Allerheiligen"
-
-        if self.subdiv in {
-            "AI",
-            "LU",
-            "NW",
-            "OW",
-            "SZ",
-            "TI",
-            "UR",
-            "VS",
-            "ZG",
-        }:
-            self[date(year, DEC, 8)] = "Mariä Empfängnis"
-
-        self[date(year, DEC, 25)] = "Weihnachten"
-
-        if self.subdiv in {
-            "AG",
-            "AI",
-            "AR",
-            "BE",
-            "BL",
-            "BS",
-            "FR",
-            "GL",
-            "GR",
-            "LU",
-            "NE",
-            "NW",
-            "OW",
-            "SG",
-            "SH",
-            "SO",
-            "SZ",
-            "TG",
-            "TI",
-            "UR",
-            "ZG",
-            "ZH",
-        }:
-            self[date(year, DEC, 26)] = "Stephanstag"
-
-        if self.subdiv == "GE":
-            self[date(year, DEC, 31)] = "Wiederherstellung der Republik"
+            # National Day.
+            self._add_holiday(tr("Nationalfeiertag"), AUG, 1)
+
+        # Christmas Day.
+        self._add_christmas_day(tr("Weihnachten"))
+
+    def _add_subdiv_holidays(self):
+        if self.subdiv and self.subdiv != "VS":
+            # Good Friday.
+            self._add_good_friday(tr("Karfreitag"))
+
+            # Easter Monday.
+            self._add_easter_monday(tr("Ostermontag"))
+
+            # Whit Monday.
+            self._add_whit_monday(tr("Pfingstmontag"))
+
+        if self.subdiv and self.subdiv not in {"GE", "JU", "NE", "VD", "VS"}:
+            # St. Stephen's Day.
+            self._add_christmas_day_two(tr("Stephanstag"))
+
+        super()._add_subdiv_holidays()
+
+    def _add_subdiv_ag_holidays(self):
+        # Berchtold's Day.
+        self._add_new_years_day_two(tr("Berchtoldstag"))
+
+    def _add_subdiv_ar_holidays(self):
```

Comment:
Not sure if we want to keep dummy subdiv methods.

Reply:
I left them just to show that we didn't accidentally miss these subdivs.